packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -417,6 +493,120 @@ export class Deployments {
     });
   }
 
+  public async rollbackStack(options: RollbackStackOptions): Promise<RollbackStackResult> {
+    const {
+      stackSdk,
+      resolvedEnvironment: _,
+      cloudFormationRoleArn,
+      envResources,
+    } = await this.prepareSdkFor(options.stack, options.roleArn, Mode.ForWriting);
+
+    if (options.validateBootstrapStackVersion ?? true) {
+      // Do a verification of the bootstrap stack version
+      await this.validateBootstrapStackVersion(
+        options.stack.stackName,
+        BOOTSTRAP_STACK_VERSION_FOR_ROLLBACK,
```

Comment:
Not sure what you mean. How does it downgrade what?

Reply:
If a future stack defines `requiresBootstrapStackVersion: 25` - this would still check for `23`, and run the rollback (i.e stack deployment), bypassing the check.

Reply:
Fair enough.